util/progress/progressui/printer.go

```diff
@@ -28,13 +30,71 @@ type lastStatus struct {
 
 type textMux struct {
 	w         io.Writer
+	printer   vertexPrinter
 	current   digest.Digest
 	last      map[string]lastStatus
 	notFirst  bool
 	nextIndex int
 	desc      string
 }
 
+type vertexPrinter interface {
+	printVtx(p *textMux, t *trace, v *vertex)
+}
+
+type vertexUpdate struct {
+	Digest   digest.Digest          `json:"digest"`
+	Name     string                 `json:"name,omitempty"`
+	Statuses []*client.VertexStatus `json:"statuses,omitempty"`
+	Warnings []client.VertexWarning `json:"warnings,omitempty"`
+	Logs     []vertexLog            `json:"logs,omitempty"`
+	Canceled bool                   `json:"canceled,omitempty"`
+	Error    string                 `json:"error,omitempty"`
+	Cached   bool                   `json:"cached,omitempty"`
+	Done     bool                   `json:"done,omitempty"`
```

Comment:
Instead of `Done` this should have the `Start/Completed` time.

Reply:
Modified this to just copy the `Completed` attribute from the `client.SolveStatus` instead of using `Done`.